### ISSUE TITLE
SIGSEGV (0xb) at pc=0x00000000000099a6 with native lua interpreter on server (linux)

### ISSUE THREAD

**Issue by phoxmeh** [MightyPirates/OpenComputers#436]
I can reproduce this a lot and it's quite inconsistent sometimes.
I'm on linux running the latest java version (also tried previous versions back to 7u55) and I get this seg fault on the server I run. It seems to mainly happen only when I'm running any other minecraft server and/or client on the same machine. As expected everything works when forcing LuaJ but when using the native interpreter it causes an inconsistent crash (works for a few days and then constantly crashes and then stops crashing and works again with the native interpreter)
I'm running it with other mods from the CrackPack on the atlauncher.
It always reports "SIGSEGV (0xb) at pc=0x00000000000099a6"

Here is a link to the hs_err I got when trying to get it working last time: http://pastie.org/private/w3mgfabfkivvqzcab8bgcg
I've tried both openjdk and oracle java, both have the same issue.

**Comment by fnuecke** [comment]
Thanks for the report.

Does this have the same stacktrace whenever it happens (i.e. the last Java frame being `com.naef.jnlua.LuaState.lua_isthread`) or does that vary?

Also, could you check the used native lib is the 'right' one (`...-native.64.so` from the looks of it)? The code to determine which one to use got so bloated I decided to just have it try each one until one works, but it may be that causes unexpected side effects... have you seen this in an OC version pre 1.3.0 / build 505?

**Comment by fnuecke** [comment]
Oh, also, to narrow down the source of the issue a bit, could you try to reproduce it after changing the following config options (ideally one by one, to narrow it down even further, but feel free to start with enabling all to see if any of them do anything) - `disableUserdata`, `disablePersistence`, `disableMemoryLimit`. As the names indicate (and the comments in the config), these lead to reduced functionality, but it'd help me a lot to get an idea of where to start looking. Thanks!

**Comment by XDjackieXD** [comment]
I got the same problem running a dedicated server on ubuntu 64-bit (computers - most of the time - crash the server with a sigfault when turned on or loaded on server start).
Enableing LuaJ fallback solves it.
I will try these settings in a minute.

Update:
Disabeling persistence seems to solve the problem... (after several restarts of the computers and the Minecraft server it crashed not a single time)

**Comment by phoxmeh** [comment]
It looks like a lot of them end at `com.naef.jnlua.LuaState.lua_newstate`. 
I've tried disabling `disableMemoryLimit` but not the other two. 
The Library seems to be the correct one loading. 
This has been an issue since 1.3 (although I didn't have any versions before that on a server)

@XDjackieXD And the persistence is my favourite features too! XD

Also disabling the persistence has not helped me. I've tried disabling userdata and the memory limit (and all at once) and I get the same result.

**Comment by XDjackieXD** [comment]
@phoxmeh yeah persistance is a really cool feature...

@fnuecke when having persistanec enabled and loading a world with a computer in the on-state the error happens at "com.naef.jnlua.LuaState.lua_newstate(IJ)V+0" (according to the hs_err_pidxxxx.log)

**Comment by fnuecke** [comment]
If this is reproducible for a certain world, would it be possible for you to send me either the `<savedir>/opencomputers/state` folder or the whole world? Maybe I can reproduce it with that and get some more info out of the crash. Thanks!

**Comment by phoxmeh** [comment]
Mine still crashes unless I force it to use LuaJ :/ which is strange since it was working perfectly fine with the native library for some time before I went to reboot the server.

I don't have any data under any of the `.../opencomputers/state` sub-directories. My current world is about 900+MB uncompressed. I could compress it to an archive you prefer (unless you don't mind a tar.7z archive since that's how I backup) and send you a copy.

**Comment by fnuecke** [comment]
> I don't have any data under any of the .../opencomputers/state sub-directories.

That's interesting, actually, since that points to something being broken in JNLua, not in Eris (which might still be my fault, ofc [/disclaimer]). The world should be pretty irrelevant then, but thanks for offering! I'll try rebooting my test server a bunch of times, will see if something pops up.

Otherwise I may have to generate some custom builds so you can test with those. We'll see. Thanks for your patience!

**Comment by XDjackieXD** [comment]
I will test it tomorrow using a clean installation of OpenComputers on Minecraft 1.6.4 (I am currently testing using the latest experimental snapshot of the yogscast pack)

And btw: In my state folder there is an empty folder called "0" but nothing else...

Update: Can't reproduce it using a fresh install of OC 1.3.2.... (Also I don't get the crash in my old world using the yogscast pack anymore.... strange....)

**Comment by XDjackieXD** [comment]
Karma hit me... :D
The second I updated my previous comment the error hit me again (Yogscast Pack OC 1.3.1.516)...
Disabeling persistence fixed it again.

http://pastebin.com/Drg60xtG

**Comment by fnuecke** [comment]
Thanks for that log. Though the persistence setting shouldn't have any influence at that point, anything's possible with segfaults :-P While I'm kind of expecting that to be by chance, I'll see what I can find. I haven't been able to reproduce it, yet, though, so it's kinda purely going through whatever I can think of mentally right now... I'll try with the full pack when I have the time, just in case.

**Comment by XDjackieXD** [comment]
I cannot reproduce it using a clean install...
Idk why but it may be related to anything in this pack.
But disabeling persistence definatly fixes the crash (reproduced it several times using the latest yogscast pack)

**Comment by fnuecke** [comment]
All right. In the worst case it's something triggered by the garbage collector running more frequently because of the higher memory load from the pack... that would be fun to debug...

**Comment by phoxmeh** [comment]
I'm using The Crack Pack, not sure what mods are shared between it and the Yogscast pack. But I do know it's very inconsistent when it decides to crash for me, albeit now it crashes always. I did find that when someone started spamming world anchors around it was crashing more often (I promptly disabled them). Also it seems to crash more reliably when I'm running both the client and server at the same time. Right now it just seems to always crash when using the native lib, if the computers are on when using the native lib it will crash, just usually quicker or immediately when I'm running my client too. Haven't tried it in a day so it might decide to work for a couple days cause that's what it seems to do for me.

**Comment by fnuecke** [comment]
Eh, it's probably less the packs but their effects on the runtime (having tons of classes loaded, using lots of memory). That's my guess, anyway. Which Linux distros are you running, by the way?

**Comment by phoxmeh** [comment]
I'm running an arch64 install.

**Comment by XDjackieXD** [comment]
I'm running Ubuntu 14.04 64bit desktop.

In my case the crash occurs about 60% of the times i turn on a Computer and everytime a chunk with a turned-on Computer gets loaded (until I disable persistence).

**Comment by Kilobyte22** [comment]
here a small suggestion: enable core dumps using `ulimit -c unlimited` on the terminal, as the log suggestes. next time it crashes you will get a file named something with core in your ~/.minecraft (for servers in the server dir). Send that file to @fnuecke. It contains debug information he can use to track down the bug. He will probably send you a debug build without stripped debug symbols in a bit, so wait for that. We already talked about that in IRC

**Comment by XDjackieXD** [comment]
https://mega.co.nz/#!OFcmASKY!XZUvz1j7jeVn8Z-p700LYMrqAwVNG51TDpDCrrsMfeY
275MB of compressed coredump goodness :D

**Comment by phoxmeh** [comment]
So my core dump is 2.9GB.... this might take a while XD
edit: so it compressed better than i thought so hopefully i should get it uploaded... wish my upload speed was better

**Comment by fnuecke** [comment]
Awesome, thanks! I'll see what I can glean from that :-)

**Comment by phoxmeh** [comment]
https://copy.com/idJj28NApPvl
sha1sum: 3b4864c45f01b3d11708b901aa4fcd884ce412dd

there is mine, hope that works XD

**Comment by fnuecke** [comment]
Thanks! I'll probably need to set up a VM with arch to make use of @phoxmeh's coredumps, but at least I have the Ubuntu VM at a point where everything "environmental" fits the coredump, from what I can see. So: slow progress, but progress.

I've been running the Yogspack on the Ubuntu VM for ~12h now, with one computer being stopped and then started again every ten seconds, but no crash yet, so I'm afraid I'll have to rely on you for helping me solve this for a little longer.

I'll have to have you both use ~~[this custom built version with debug libs](http://cil.li/mc/OpenComputers-MC1.6.4-1.3.3-dev-universal.jar)~~ to get anything useful out of the dumps, though. Please switch out the OC JAR in the packs with this one and get me a coredump, that _should_ tell me a little bit more then! This was built on the Ubuntu VM, so I'd primarily need @XDjackieXD to get me a coredump with this. @phoxmeh you could at least verify it still happens with that build, you can probably save yourself the time and bandwidth of uploading a coredump until I get an arch VM up and running :-P

**Obligatory warning**: the custom build is based on the dev branch, so make sure you have a backup of your world, just in case.

Thanks again for your patience and cooperation :-)

**Comment by XDjackieXD** [comment]
https://mega.co.nz/#!OVkVwRxZ!q_phLqlZ7pc__oPCtNIYTF6nAF3yaQTMaaErT5_bbao

Here you have the coredump and error log (wich gives a little more detail this time :) )
At least the crash is reproducable over different versions of oc...

**Comment by phoxmeh** [comment]
I did it anyways :P
https://copy.com/4IruU8pIGhgH
sha2sum: d67280f8e2403b41856c2a544bec60af94ff63b1

**Comment by XDjackieXD** [comment]
No idea what I changed on my setup (could a kernel update have something to do with it?) but now the crash can only prevented by enabling LuaJ fallback...

**Comment by fnuecke** [comment]
Hmm, maybe? I had a look at the debug dump, and I'm afraid it's not _that_ helpful :-( The crash itself, from what I can tell, happens inside some magic code that takes care of thread-local variable assignment (it happens [on this line](https://github.com/fnuecke/jnlua/blob/eris/jnlua/src/main/c/jnlua.c#L849), which is [really just this](https://github.com/fnuecke/jnlua/blob/eris/jnlua/src/main/c/jnlua.c#L31); all stack frames after that line are just question marks, even in the debug build).

The main hindrance for me is that I still can't reproduce it, so I can't try to catch the library screwing up red-handed. But I'll keep digging and let you know when I have some new thing to test out.

Actually, one thing you might try, since it seems to be related to thread-local storage at least in some capacity, is to set the number of worker threads to one (from the default four). Yes, that's just a wild guess...

**Comment by Wuerfel21** [comment]
As when speaking of mamy classes, could raising Permgen help?

**Comment by fnuecke** [comment]
Very unlikely.

**Comment by XDjackieXD** [comment]
Switching it to 1 worker thread seems to solve the problem after several server restarts and half an hour playing...

**Comment by fnuecke** [comment]
@XDjackieXD to clarify, it didn't crash at all across these restarts? Or it didn't crash after the most recent restart?
@phoxmeh could you see if this also applies in your case (i.e. setting worker thread count to 1 avoids the crash)?

If this does indeed help, it means a bit of work, but at least I'd have something to try out.

**Comment by XDjackieXD** [comment]
It doesn't crash at all after setting it to 1 (I played about 1h and made a few restarts with 2 computers turned on and I switched them on and of and nothing crashed)

**Comment by phoxmeh** [comment]
Yep, absolutely no crash so far. I'll keep it going for a while, some people should get on a bit more later and if it doesn't crash I'll let ya know but so far it's working fine.

**Comment by fnuecke** [comment]
Fingers crossed. At least it's a non-catastrophic workaround for now. I'll dig some more through the JNLua code over the weekend and see if I can maybe get rid of the thread-local stuff altogether in a hope of solving this.

**Comment by fnuecke** [comment]
All right, got rid of the last use of thread-local variables, had it running over the night with no issues to make sure I didn't mess up something else. @XDjackieXD, @phoxmeh please give [this version a try](http://cil.li/mc/OpenComputers-MC1.6.4-1.3.3-dev-universal.jar) and let me know how it goes. Don't forget to up the threads to 4 again. Thanks!

**Comment by XDjackieXD** [comment]
Im not at home for 2 weeks so I wont be able to test until sunday in 2 weeks.
Also I got a new pc on friday so I have a second computer with ubuntu 64-Bit to test with (and a lot faster than in my laptop ;D ).

**Comment by phoxmeh** [comment]
~~Well it's not crashing now (after remembering to turn off the computers before starting up with the native lib and threads set to 4) but now it's telling me `oc:native libraries not available` with http://pastie.org/private/qnvbcggte7whxavltmpw in console~~

edit: ignore that... after a little tinkering it works fine...

**Comment by Kilobyte22** [comment]
@XDjackieXD well, on a different machine it might behave differently. @fnuecke wasn't able to reproduce it on Ubuntu himself afaik

**Comment by fnuecke** [comment]
@phoxmeh hmmm. Could you post the full log / some more context around it, just in case?

**Comment by phoxmeh** [comment]
The main thing i did was remove the computers before restarting the server again. Once I did that it worked just fine. Nothing else was reported in the logs as far as I know but I'll be more than glad to upload that forge log if you need it.

**Comment by fnuecke** [comment]
Oh, OK. Hmm, keep an eye on it, then, please. If after another re-start the above happens again, let me know!
(And I'll have another look over the changes I made to see if anything could cause that.)

**Comment by phoxmeh** [comment]
Well it seems to be failing still @_@
it crashed with this http://pastie.org/private/0k5ipygblrafwghwdkplq
and this http://pastie.org/private/xfytvaxg3bbb8j9thyvtxw
so far

**Comment by fnuecke** [comment]
Darn. Well, it was an expected possibility. If you can, would you mind getting me a coredump for this latest build? I'll finish setting up the arch VM in the evening, hopefully I'll be able to get anywhere with that.

Also, just to make sure, the debug settings mentioned above (`disable*`) still don't have any influence on whether it's crashing or not?

**Comment by phoxmeh** [comment]
Ok, gonna try to test it out a bit more, sorry I've been quite a bit busy at work and unable to really do any minecraft stuff the past couple days. Currently I'm running the dev version with nothing disabled right now and threads at 4, no crashes yet except for the library failing to load when i first chagned it to the dev version and a small crash when i was shutting it down. So far nothing, gonna try some restarts and play around with it. I'll let ya know how it's running tomorrow.

**Comment by phoxmeh** [comment]
So I've been testing the server a bit and it has been stable with no crashes and I just went to restart it and it started crashing again like the last time I had issues (running the dev version you gave me). Tried diabling things and setting threads to 1 but that all failed. Only after restarting with LuaJ forced and removing the computers and restarting twice (cause the first time it doesn't load the native lib properly) does it seem to work with everything enabled and threads at 4.
The computers are left on when I shut down the server initially, but it doesn't always crash just usually after the server has been up a couple days. During the uptime it's perfectly fine but when I try to start it again it crashes with the same segfault as http://pastie.org/private/0k5ipygblrafwghwdkplq
I'll try to reporduce it again and get a core dump (always forget to enable it to get that before I do anything @w@)
Lemme know if there's anything else you want me to try and do to figure this out. I'll keep tinkering how I have been to see if i can get that core dump for you.

**Comment by fnuecke** [comment]
Thanks so much for taking the time to test all this! I'm afraid I didn't have the time to test as much over the weekend as I hoped I'd be able to, but someone brought up an interesting issue that _might_ be related (garbage collection, which might explain why it only happens for existing / resumed computers). I'll investigate this as soon as possible, probably tomorrow.

Side note: if we can't find a robust solution/workaround for this in the next couple of days I'll probably still push out 1.3.3 with this as a pending known issue, due to all the other fixes that have accumulated...

**Comment by fnuecke** [comment]
@phoxmeh when you have the time, could you please give [the latest dev build](http://ci.cil.li/job/OpenComputers-dev-MC1.6.4/) a try? I'm disabling the Lua GC while persisting / unpersisting now, since that reportedly helped with the other issue I mentioned, so that _may_ help with this, but may just as well be unrelated.

**Comment by phoxmeh** [comment]
Sorry it's taking me so long work on things, it's been quite the busy week. 
I got the lately build today, 601 I believe, and it's crashing regardless of settings unless I force LuaJ. I'd removed the computers before doing anything so to make sure they were off and not saved in the world as to casue any problems. But it's crashing the same as it was in the beginning and at the same adress space (0x0...99a6)
That last dev build you sent me was working flawlessly until I had to do reboots. I'll keep trying to see if I can get it to launch the computers with the current build though. Just lemme know what else you need me to do.

here is the hs_err: http://pastie.org/private/a727keh23unsuwswxnxhpa

**Comment by fnuecke** [comment]
No problem, thanks for helping out with this! All right, so it's most likely not related to that other issue. I guess that's kind of good. But it doesn't really get us closer to a solution, either. Hmm. Which version of libc do you have installed? (`ldd --version`)

**Comment by phoxmeh** [comment]
GNU libc 2.19

**Comment by fnuecke** [comment]
Hmm, so the exact same one I have, so it's not that, either. Gah. Back to the drawing board.

**Comment by fnuecke** [comment]
In the [latest **dev** build](http://ci.cil.li/), could you try setting the config option `computer.debug.forceNativeLibWithName` to "native.64.so", to ensure it's the only one it's trying to load, to ensure that trying to load possibly incompatible libs doesn't somehow screw over the state, leading to these issues? Thanks!

**Comment by XDjackieXD** [comment]
I used the latest dev build and created a new world (to prevent a false crash because of a corrupt world or such) and put 'forceNativeLibWithName="native.64.so"' in the debug section of the config and exactly the same crash happens (SIGSEGV (0xb) at pc=0x00000000000099a6, pid=4584, tid=140127246993152).
http://pastebin.com/8X6VbaMZ <- crash log
Shall I also upload the coredump?

**Comment by fnuecke** [comment]
Thanks. Well, at least that means it's not the auto-detection, which is kinda good, I guess.

I doubt the dump would tell me anything new, in particular since that lib is without the debugging symbols, but thanks for offering. I'll wrap up another build with some JNI warnings that popped up when running JNLua's test suite with `-Xcheck:jni` patched away. I'm skeptical it'll help, but it's worth a shot.

Aside from that, could those that see this crash please let me know which CPU they have? E.g. pastebin your `/proc/cpuinfo`? Seeing as this only appears to happen on a pretty small number of systems, I wouldn't even be surprised if it's somehow hardware related (although that would be a major pain).

**Comment by XDjackieXD** [comment]
My laptop has a  Core i7 3612QM and my new PC has a i7-4790k
Thats the cpuinfo of my pc: http://pastebin.com/sjREm9b7

**Comment by Doyan** [comment]
Hi I dropped by the irc yesterday, I seem to have the same problem. When a computer is created for the first time my server is crashed, if i try to log in at once and load that chunk again it sometimes works for a few logins but the next day or so the server will crash on login untill I force luaj. Sometimes both the client and the server crashes on login and sometimes only the client crashes. But it always crashes when it has gotten to that point until I force luaj. 

I'm running the server with a private modpack on my own computer running crunchbang waldorf (based on debian wheezy). logs follows below:

hs_err:
http://pastebin.com/qhGTf1cW 

forge serverlog:
http://tny.cz/13e1de01

proc/cpuinfo
http://pastebin.com/uK0azx8n

**Comment by XDjackieXD** [comment]
I don't get client crashes but on the Server side you definatly got the same crash...

At least we now know that its not Intel only (my first thought) and it happens only in 64bit linux systems (or is anyone here who can test in a 32bit Linux?)

Edit: hate the german auto correction when writing in English...

**Comment by fnuecke** [comment]
All right, dev build 614 has the new libs in it, please give it a go when you have the time. As I said, I'm not very optimistic, but it's worth a shot.

> client side crashes

Are they the same crash / related? Because that would be... kinda odd. I'd be interested in a stack trace of those.

And yeah, if it happens on AMD and Intel CPUs we can _probably_ scrap that route of investigation.

Edit: hmmm, another random thought, how much RAM do you have? Maybe it's a pointer getting truncated to 32 bit somewhere... that'd at least explain why it only happens on 64 bit systems.

**Comment by Doyan** [comment]
I gave your latest dev build a try just now, build 620 if I read your jenkins correct. This all feels fairly inconsistent. And I don't know at all enough to understand any of it so I'll just recollect my experiences the way I would at the doctor...

At first it would let me login but my computer would be off and any attempt to start it would be greeted with the message "oc:native libraries not available" 

After 3 server restarts i was able to turn my computer on and everything seemed to work just fine. At this point I was able to restart the client and relog a couple of times, persistence working as it should. 
I was even able to restart the server, everything working as intended. 

Since I wanted to make sure my friends could login aswell I updated my modpack.zip and then updated my client through the Technic launcher.

When I restarted the after updating my client through the Technic launcher my client crashed on login, when i started the client again and logged in i was greeted with the "oc:native libraries not available" message again. After a second relog persistence appeared to be working again as if the crash never happened.

When all seemed to work as intended I did a server restart, on login after that my computer was off but I was able to start it with no error messages albeit without any previous written lines showing up. when relogging everything seemed to work as intended.

After that I restarted my client, on login after client restart my tier 2 screen had the interface of a tier 1 and it wouldn't let me type on it. This was fixed after a relog. I was able to repeat this pattern two times by restarting the client. 

As of now I have restarted the server twice with everything working as it should, client restarts included.

hs_err from client crash

http://tny.cz/ed6c6699 

forge server log when client crash occurred and subsequent relogs first showing the oc:native libraries not available message and then working as it should.

http://tny.cz/d7ff7a92

my proc/meminfo and my proc/cpuinfo should be at the bottom of the hs_err.

**Comment by Doyan** [comment]
As I said I don't know at all enough to understand any of this, so I apologize for anything being irrelevant. I will probably check again tomorrow or so to see if it is still working.

**Comment by XDjackieXD** [comment]
I just tested build 614 and it doesn't crash but I can't start a computer and the server log shows following error:

```
2014-08-25 10:02:53 [WARNUNG] [OpenComputers] Failed creating Lua state.
java.lang.IllegalStateException: Lua state is closed
    at com.naef.jnlua.LuaState.check(LuaState.java:2311)
    at com.naef.jnlua.LuaState.<init>(LuaState.java:260)
    at com.naef.jnlua.LuaState.<init>(LuaState.java:250)
    at li.cil.oc.util.LuaStateFactory$.createState(LuaStateFactory.scala:211)
    at li.cil.oc.server.component.machine.NativeLuaArchitecture.initialize(NativeLuaArchitecture.scala:279)
    at li.cil.oc.server.component.machine.Machine.init(Machine.scala:706)
    at li.cil.oc.server.component.machine.Machine.start(Machine.scala:129)
    at li.cil.oc.server.PacketHandler.trySetComputerPower(PacketHandler.scala:60)
    at li.cil.oc.server.PacketHandler.onComputerPower(PacketHandler.scala:43)
    at li.cil.oc.server.PacketHandler.dispatch(PacketHandler.scala:22)
    at li.cil.oc.common.PacketHandler.onPacketData(PacketHandler.scala:26)
    at cpw.mods.fml.common.network.NetworkRegistry.handlePacket(NetworkRegistry.java:255)
    at cpw.mods.fml.common.network.NetworkRegistry.handleCustomPacket(NetworkRegistry.java:245)
    at cpw.mods.fml.common.network.FMLNetworkHandler.handlePacket250Packet(FMLNetworkHandler.java:85)
    at net.minecraft.network.NetServerHandler.func_72501_a(NetServerHandler.java:1130)
    at net.minecraft.network.packet.Packet250CustomPayload.func_73279_a(SourceFile:61)
    at net.minecraft.network.TcpConnection.func_74428_b(TcpConnection.java:462)
    at net.minecraft.network.NetServerHandler.func_72570_d(NetServerHandler.java:141)
    at net.minecraft.network.NetworkListenThread.func_71747_b(NetworkListenThread.java:54)
    at net.minecraft.server.dedicated.DedicatedServerListenThread.func_71747_b(SourceFile:30)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:691)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:276)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:587)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:484)
    at net.minecraft.server.ThreadMinecraftServer.run(SourceFile:583)
```

and in chat it says that the native lib is not available (also the latest dev build 622 and setting the debug config mentioned earlier doesn't change anything)

**Comment by fnuecke** [comment]
@Doyan thanks, that's very interesting indeed! I _much_ prefer a little bit of potentially irrelevant info over "it no work, fix plz" ;-) In particular the client log is very intriguing, though I don't want to draw any quick conclusions from that just yet.

@XDjackieXD thanks, that seems to be the same effect present in Doyan's log. It seems something breaks after the first "test" state is created; possibly in the close/garbage collection logic messing up. I'll see what I can find there.

For most people it doesn't seem to have a side effect, but could you both try setting `debug.forceNativeLibWithName="native.64.so"` in the config (wasn't mentioned in here before as far as I can tell; if you did try that, double check please, it was moved from the `computer.debug` category to just `debug` in the latest builds).

Other than that I'll have a closer look at a few pieces of code that I think might be related to this behavior as soon as I can.

**Comment by Doyan** [comment]
Yesterday when I started the server with the above config option set it threw the "oc: native libraries not available" I didn't bother restarting then cause people was on. 

Today when I started it threw the same error, I then restarted the server. After restart I was able to start my computer without error messages albeit without any previous state loaded.

**Comment by fnuecke** [comment]
OK, sort of expected. Thanks. If I understand correctly, with the latest builds the (server-side) errors/warnings are non-fatal, though, yes?

I made one more alteration, when you have the time, please see how builds 630/115+ behave. I'm afraid after that that I'm out of ideas again, for the time being.

**Comment by Doyan** [comment]
Yes, the errors and varnings are non fatal, the last times I've started the server the pattern has been the same, after a server restart the mod works as expected altough with no previous state loaded. 

The next day it is the same, first time I start "oc: native libraries not available" and after a restart or so it works like stated above. It seems like it needs to be started twice before it can load the state. Or delete it to make a new I don't know how it works.

Havent gone many days though so this might just be coincidence...

I'll try the next build sometime tomorrow hopefully.

**Comment by Doyan** [comment]
I tried out the latest build today, when I first logged on and started the computer there was a short lagspike but it started correctly. written lines persisted through a server restart. 

When I tried a client restart after that I had the same issue mentioned before with my tier 2 screen behaving like a tier 1 and not letting me input. As before a relog fixed that, persistence working as it should.

So far it looks good! I wont be able to do any extensive restarting/trying different stuff today. But this looks promising.

**Comment by fnuecke** [comment]
Haha, don't get my hopes up ;-) But yeah, sounds promising. @XDjackieXD, @phoxmeh if you would, could you give the latest dev builds a try, too, to either confirm or invalidate? Thanks.

As for the screen issue, that sounds like something is lost while synchronizing the client... while annoying, that should be purely 'cosmetic', I'll look into that when I can. Feel free to open a new issue on that, if it happens again (maybe see if there are any warnings from OC in the log).

Oh, and the lag spike _probably_ stems from the "running" sound being loaded the first time... I keep forgetting looking into preloading that...

**Comment by phoxmeh** [comment]
I'll try to test some things soon, been swamped with work plus my internet went out for about a week due to some lines getting messed up @w@
I might be able to try things out by Sunday when things get calmer, I really want to play some minecraft XD

**Comment by Wuerfel21** [comment]
Why noone uses a debugger/tracer to see when it's happening?

**Comment by fnuecke** [comment]
@Wuerfel21, if you can reproduce this and are capable of debugging a native library that's been loaded into Java, then by all means, do. Otherwise I recommend you read the above posts: I cannot reproduce this, but coredumps were made, not much of use was to be learned from them, sadly.

**Comment by fnuecke** [comment]
No feedback in over a week, if this is still an issue in the latest version let me know and I'll re-open this.

Fingers crossed.

**Comment by phoxmeh** [comment]
Sorry for the lack of feedback. I've not gotten to try it too much yet. On my 1.6.4 server I had no issue with the 1.3.4 patch but didn't test it extensively. Moving to my 1.7.10 server now things seem fine too. I'll report if I encounter the issue again. Thank you for all your hard work in making this amazing mod!
